pydantic/_internal/_generate_schema.py

```diff
@@ -317,6 +466,8 @@ def _generate_schema_for_type(
             if metadata_schema:
                 self._add_js_function(metadata_schema, metadata_js_function)
 
+        _add_custom_serialization_from_json_encoders(self._json_encoders, obj, schema)
```

Comment:
Should this line go above the `metadata_js_function` stuff? It seems if this line is going to modify the serialization, it would make sense to generate the JSON schema after that modification to `schema` has been performed.

Reply:
Does it matter? We're editing the CoreSchema here directly while the js func won't get executed until later. Or maybe I'm misunderstanding. Is there a test case we can add?